src/compiler/checker.ts

```diff
@@ -16549,6 +16559,34 @@ namespace ts {
             return unknownType;
         }
 
+        function getTypeOfArrayLiteralOrObjectLiteralDestructuringAssignment(expr: Expression): Type {

```

Comment:
Is `expr` the initializer? Or the assignment expression? Can you document that explicit with a comment and using a slightly more precise type than `Expression`?